### PR TITLE
Pyic 7042: spike UI tests

### PR DESCRIPTION
**Pull request by shivanshuit914**
## Proposed changes

- Add UI snapshot testing example
- Add accessibility testing example
- Add docker pipeline test run
- Add imposter mock server with steps tests

### What changed

- New UI tests using playwright
- Did not use cucumber because we don't want to test journeys.

### Why did it change

- To test UI.

### Issue tracking

- [PYIC-7042](https://govukverify.atlassian.net/browse/PYIC-7042)

[PYIC-7042]: https://govukverify.atlassian.net/browse/PYIC-7042?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/browser/package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "ipv-core-front-test",
+  "version": "1.0.0",
+  "description": "",
+  "scripts": {
+    "test": "npx playwright test --update-snapshots"
```
Comment (shivanshuit914) [inline test/browser/package.json:6]: Need to find better way to updates snapshots

File: test/browser/accessibility.spec.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const { test, expect } = require("@playwright/test");
+const AxeBuilder = require("@axe-core/playwright");
+
+test.describe("accessibility test", () => {
+  test("should not have any automatically detectable accessibility issues", async ({
+    page,
+  }) => {
+    await page.goto("/dev/template/page-ipv-identity-document-start/en");
+
+    const accessibilityScanResults = await new AxeBuilder({ page })
```
Comment (blakeyp) [inline test/browser/accessibility.spec.js:10]: Looks good, so we should be able to iterate `all-templates` to scan each template

File: test/browser/journey.spec.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "Prove your identity at a Post Office with one of the following types of photo ID – GOV.UK One Login",
+  );
+  await page.click("input[type='radio'][value='end']");
+  await page.click("button[id='submitButton']");
+  await page.waitForURL("**/pyi-escape");
+  const titleEscape = await page.title();
+  expect(titleEscape).toBe(
+    "Find another way to prove your identity – GOV.UK One Login",
+  );
+});
```
Comment (blakeyp) [inline test/browser/journey.spec.js:19]: Nice 🎉 we'll probably want to do some context and lang checking in these types of tests too

**Review by blakeyp — COMMENTED** [review]
Looks really good 🚀

**Comment by blakeyp** [comment]
Holding PR - just a spike

**Comment by Joe-Edwards-GDS** [comment]
Can this be closed now?

**Pull request closed without merging**